src/components/MetadataResolver.tsx

```diff
@@ -1,30 +1,26 @@
 import React, { Component, ComponentClass } from 'react';
 import Client from '../services/ApolloClient';
-import { match } from 'react-router';
+import { RouteComponentProps } from 'react-router';
 
 interface State {
     metadata: any;
     data: any;
     loading: boolean;
 }
 
-interface RouteProps {
-    match: match<any>;
-}
-
 export default function MetadataResolver<P>(metadataQuery: Function, dataQuery?: Function) {
     return (WrappedComponent: ComponentClass<P>) => {
-        return class MetaDataQueryResolver extends Component<P & RouteProps, State> {
+        return class MetaDataQueryResolver extends Component<P & RouteComponentProps<any>, State> {
             defaultState: State;
             onlyMetadata: boolean;
             noQuery: boolean;
 
-            static selectQuery (props: Readonly<P & RouteProps>, state: State, isMetadataQuery: boolean) {
+            static selectQuery (props: Readonly<P & RouteComponentProps<any>>, state: State, isMetadataQuery: boolean) {
                 if (isMetadataQuery && typeof metadataQuery === 'function') {
-                    return metadataQuery({ ...state, match: props.match });
+                    return metadataQuery({ ...state, match: props.match, location: props.location });
 
                 } else if (typeof dataQuery === 'function') {
-                    return dataQuery({ ...state, match: props.match });
+                    return dataQuery({ ...state, match: props.match, location: props.location });
```

Comment:
inject all 3 route props (match, location, history) for the sake of conformance with `RouteComponentProps`